plutus-pab/test-node/README.md

```diff
@@ -0,0 +1,159 @@
+## Instructions for running a public testnet end-to-end integration test
+
+What follows is a series of steps to demonstrate running an end-to-end
+integration test of the PAB on the public testnet.
+
+#### Process
+
+Assumptions:
+
+- You have the `plutus-apps` repo checked out at `$PLUTUS`
+- You have [jq](https://stedolan.github.io/jq/)
+- You have a few hours to wait for everything to sync (the first time.)
+
+Steps:
+
+You will need ~6 terminals to run all the different components.
+
+
+1. Build all the necessary components:
+
+  ```
+  > cd $PLUTUS
+  > cabal build plutus-pab-examples plutus-chain-index cardano-node cardano-wallet
+  ```
+
+2. Start the testnet node locally:
+
+  ```
+  > cd $PLUTUS/plutus-pab/test-node/
+  > ./start-testnet-node.sh
+  ```
+
+  Note: It's important that this is done from the specific directory, because
+  the config paths are relative.
+
+3. Run the cardano wallet backend:
+
+  ```
+  > cd $PLUTUS/plutus-pab/test-node/
+  > cabal exec -- cardano-wallet serve \
+      --testnet testnet/testnet-byron-genesis.json \
+      --node-socket testnet/node.sock
+  ```
+
+4. Create or restore a public testnet wallet:
+
+  Here you have a few options. You can re-use an existing wallet, or create
+  one.
+
+  **Create a new wallet**
+
+  - Generate a recovery phrase:
+
+  ```
+  > cabal exec -- cardano-wallet recovery-phrase generate
+  ```
+
+  - Save this in a file named `restore-wallet.json` in the `testnet` folder:
+
+  ```
+  > cd $PLUTUS/plutus-pab/test-node/
+  > cat testnet/restore-wallet.json
+  { "name": "PAB testing wallet"
+  , "mnemonic_sentence": ["word1", "word2", ...]
+  , "passphrase": "pab123456789"
+  }
+  ```
+
+  - Load the wallet:
+
+  ```
+  > cd $PLUTUS/plutus-pab/test-node/
+  > curl -H "content-type: application/json" -XPOST \
+    -d @testnet/restore-wallet.json \
+    localhost:8090/v2/wallets
+  ```
+
+  Then, take note of the Wallet ID that is created, and save that:
+
+  ```
+  > export WALLET_ID=...
+  ```
+
+  - Put some tAda in it, by first obtaining an address:
+
+  ```
+  > curl -H "content-type: application/json" \
+      -XGET localhost:8090/v2/wallets/$WALLET_ID/addresses | jq '.'
+  ```
+
+  (Any address will do.)
+
+  - Then, navigate over to the [tesnet
+  faucet](https://testnets.cardano.org/en/testnets/cardano/tools/faucet/) and
+  send it some tAda.
+
+  You can verify that the transfer has taken place by clicking the link the
+  faucet application gives you; you see a transaction like so:
+
+  ![](faucet-example.png)
+
+5. Start the chain index:
+
+  ```
+  > cd $PLUTUS/plutus-pab/test-node/
+  > cabal exec -- plutus-chain-index --config testnet/chain-index-config.json start-index
```

Comment:
I'm getting a `testnet/chain-index-config.json` not found.

Reply:
@koslambrou It should be there; did you run that command from the right dir? (note the file is added in this diff)

Reply:
Oops, I was on the wrong branch